_episodes/04-networks-are-like-onions.md

```diff
@@ -0,0 +1,529 @@
+> > That means that the remainder of the shape, namely (32, 32, 3), denotes
+> > the dimension of one image. The last value 3 is typical for color images,
+> > and stands for the three color channels r,g,b.
+> > That means the width and height of our images are both 32 pixels.
+> >
+> > We can find out the range of values of our input data as follows:
+> > ~~~
+> > train_images.min(), train_images.max()
+> > ~~~
+> > {: .language-python}
+> > ~~~
+> > (0, 255)
+> > ~~~
+> > {: .output}
+> > So the values of the three channels range between 0 and 255.
+> >
+> > Lastly, we inspect the dimension of the labels:
+> > ~~~
+> > train_labels.shape
+> > ~~~
+> > {: .language-python}
+> > ~~~
+> > (5000, 1)
+> > ~~~
+> > {: .output}
+> > So we have, for each image, a single value denoting the label.
+> > To find out what the possible values of these labels are:
+> > ~~~
+> > train_labels.min(), train_labels.max()
+> > ~~~
+> > {: .language-python}
+> > ~~~
+> > (0, 9)
+> > ~~~
+> > {: .output}
+> > The values of the labels range between 0 and 9, denoting 10 different classes.
+> {: .solution}
+{: .challenge}
+
+
+The train set consists of 50000 images of 32x32 pixels, with 3 channels (r,g,b values). The r,g,b values are between 0 and 255. For input of Neural networks, it is better to have small input values. So we normalize our data between 0 and 1:
+
+
+~~~
+train_images = train_images / 255.0
+test_images = test_images / 255.0
+~~~
+{: .language-python}
+
+The labels are single numbers denoting the class:
+We map the class numbers back to the class names, taken from the documentation:
+
+~~~
+class_names = ['airplane', 'automobile', 'bird', 'cat', 'deer',
+               'dog', 'frog', 'horse', 'ship', 'truck']
+~~~
+{: .language-python}
+
+
+Now we can plot a sample of the training images, using the `plt.imshow` function.
+~~~
+import matplotlib.pyplot as plt
+plt.figure(figsize=(10,10))
+for i in range(25):
+    plt.subplot(5,5,i+1)
+    plt.imshow(train_images[i], cmap=plt.cm.binary)
+    plt.axis('off')
+    plt.title(class_names[train_labels[i,0]])
+plt.show()
+~~~
+{: .language-python}
+
+
+![Output of plotting sample](../fig/04_cifar10.png)
+
+## Convolutional layers
+In the previous episodes, we used 'fully connected layers' , that connected all input values to all hidden nodes. This results in many connections, and thus weights to be learned, in the network. Note that our input dimension is now quite high (even with small pictures), we have:
+
+
+~~~
+dim = train_images.shape[1] * train_images.shape[2] * train_images.shape[3]
+print(dim)
+~~~
+{: .language-python}
+~~~
+3072
+~~~
+{: .output}
+
+> ## Number of parameters
+>
+> Suppose we create a Dense (fully connected) layer with 100 hidden units as input units, how many parameters does this layer have?
+>
+> > ## Solution
+> >
+> > Each of the *dim* inputs is connected with 100 hidden units, and each of the hidden units has a bias. So we have *dim * 100 + 100* weights to learn.
```

Comment:
`dim` falls a bit from heaven here. Where does this come from? Also, the solution mixes a lot of technical lingo: 
- hidden units
- bias
- parameters
- weights
- inputs

```suggestion
> > Each entry of the input dimensions, i.e. the `shape` of one signal datum, is connected with 100 neurons of our hidden layer, and each of these neurons has a bias term associated to it. So we have `307300` parameters to learn.
```